### PR TITLE
Update opentripplanner deps; use default getCompanyIcon and add as prop

### PR DESCRIPTION
**Pull request by landonreed**
- Update internal deps (so we can use `core-utils@3.0.4`)
- Fix opentripplanner/otp-ui#232.
- New stories for leg and mode icons

~Blocked until #233 can be merged/released.~

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
An updated `yarn.lock` should be checked in. I am not able to run storybook without regenerating `yarn.lock` first.

File: packages/icons/src/index.story.js [hunk 1]
```diff
@@ -9,22 +9,37 @@ function Container({ children }) {
...
 
-Object.keys(Icons).forEach(iconKey => {
-  if (noStoryComponents.indexOf(iconKey) > -1) return;
-  const Component = Icons[iconKey];
-  allStories.add(iconKey, () => (
-    <Container>
-      <Component />
-    </Container>
-  ));
+const [walkLeg, bikeRentalLeg] = bikeRentalItinerary.legs;
```
Comment (binh-dam-ibigroup) [inline packages/icons/src/index.story.js:14]: It seems the added code for testing the rendering of rental/hail legs should be in its own `.story.js` file and create a new story heading such as `storiesOf("Icons in bike rental legs", module);` or something similar.

**Comment by binh-dam-ibigroup** [comment]
> An updated `yarn.lock` should be checked in. I am not able to run storybook without regenerating `yarn.lock` first.

Never mind, disregard this comment.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please address https://github.com/opentripplanner/otp-ui/pull/234#discussion_r562800809

File: packages/icons/src/index.story.js (hunk 1 above)
Comment (landonreed) [reply packages/icons/src/index.story.js:14]: Yea, you're probably right.

File: packages/icons/src/index.story.js [hunk 2]
```diff
@@ -9,22 +9,16 @@ function Container({ children }) {
...
-
-Object.keys(Icons).forEach(iconKey => {
-  if (noStoryComponents.indexOf(iconKey) > -1) return;
-  const Component = Icons[iconKey];
-  allStories.add(iconKey, () => (
+Object.keys(Icons).forEach(key => {
+  // Skip leg and mode icons.
+  const isLegIcon = key.indexOf("LegIcon") !== -1;
+  const isModeIcon = key.indexOf("ModeIcon") !== -1;
+  if (isLegIcon || isModeIcon) return;
```
Comment (binh-dam-ibigroup) [inline packages/icons/src/index.story.js:16]: I think this file should be reverted.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Let's revert `icons/src/index.story.js`.

**Review by binh-dam-ibigroup — APPROVED** [review]
Thanks for the changes.

File: packages/icons/src/index.story.js [hunk 3]
```diff
@@ -9,22 +9,15 @@ function Container({ children }) {
...
-  const Component = Icons[iconKey];
-  allStories.add(iconKey, () => (
-    <Container>
-      <Component />
-    </Container>
-  ));
-});
+Object.keys(Icons)
+  // Skip leg and mode icons.
+  .filter(key => key.indexOf("LegIcon") > -1 && key.indexOf("ModeIcon") > -1)
```
Comment (binh-dam-ibigroup) [inline packages/icons/src/index.story.js:14]: My bad :( indexOf() === -1

**Pull request merged**